### PR TITLE
dynamic-references updated for Cloud9

### PR DESCRIPTION
**Pull request by cjhar**
## What does this PR do and why?

Updated instructions to use Cloud9 dev environment

## PR acceptance checklist

This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [ ] Added/Updated documentation if applicable.
- [X] Pre-commit checks passed.
- [X] Lint and Nag checks passed.
- [ ] If releasing a new version, have you bumped the version `make version part=<major|minor|patch>`?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by cjhar** [comment]
Updated to use CLI for Cloud9

**Comment by cjhar** [comment]
Timing was ~30 minutes waiting for RDS cfn creation. This lesson doesn't have an estimated timings though.

**Tested end2end**

File: content/intermediate/templates/dynamic-references/index.md
Comment (sdangol) [inline content/intermediate/templates/dynamic-references/index.md:1]: Indentation issue in line [21](https://github.com/aws-samples/cfn101-workshop/pull/211/files#diff-39846b770c89aa9aa10da824def06bde08f862a9b67bff7183e9df5a1059de44R21)

File: content/intermediate/templates/dynamic-references/index.md [hunk 1]
```diff
@@ -47,7 +51,15 @@ Let’s get started! Choose to follow steps shown next:
...
+:::code{language=json showLineNumbers=false showCopyAction=false}
+{
+    "Version": 1,
+    "Tier": "Standard"
+}
+:::
+
+4. Follow steps shown next to create a dynamic reference to your parameter in an EC2 instance you describe in a template:
 
     1. change directory to `code/workspace/dynamic-references`.
```
Comment (sdangol) [inline content/intermediate/templates/dynamic-references/index.md:64]:
```suggestion
    1. Change directory to `code/workspace/dynamic-references`.
```

File: content/intermediate/templates/dynamic-references/index.md
Comment (sdangol) [inline content/intermediate/templates/dynamic-references/index.md:1]: Capitalize the sentences in [138 and 139](https://github.com/aws-samples/cfn101-workshop/pull/211/files#diff-39846b770c89aa9aa10da824def06bde08f862a9b67bff7183e9df5a1059de44R138-R139)

File: content/intermediate/templates/dynamic-references/index.md [hunk 2]
```diff
@@ -131,21 +195,43 @@ Let’s get started! Choose to follow steps shown next:
...
+    ::::
+    :::::
 
    In the template you just used, database connection parameters are retrieved during stack runtime using a dynamic string. You retrieved the value for a given key, such as `RDS_HOSTNAME`, with: `'{{resolve\:secretsmanager\:DatabaseConnParams\:SecretString\:RDS_HOSTNAME}}'`, where `DatabaseConnParams` is the secret ID.
 
    ::alert[A secret in AWS Secrets Manager has [*versions*](https://docs.aws.amazon.com/secretsmanager/latest/userguide/getting-started.html#term_version) holding copies of the encrypted secret value. When you change the secret value, Secrets Manager creates a new version. A secret always has a version with the staging label `AWSCURRENT`, which is the current secret value. If required, you can modify this string specifying a *version-stage* or *version-id* as such: `'{{resolve\:secretsmanager\:prod-DatabaseConnParams\:SecretString\:RDS_HOSTNAME:<version-stage>:<version-id>}}'`. When you do not specify a version, CloudFormation defaults to resolving the secret associated with the stage `AWSCURRENT`.]{type="info"}
```
Comment (sdangol) [inline content/intermediate/templates/dynamic-references/index.md:232]:
```suggestion
   ::alert[A secret in AWS Secrets Manager has [*versions*](https://docs.aws.amazon.com/secretsmanager/latest/userguide/getting-started.html#term_version) holding copies of the encrypted secret value. When you change the secret value, Secrets Manager creates a new version. A secret always has a version with the staging label `AWSCURRENT`, which is the current secret value. If required, you can modify this string specifying a *version-stage* or *version-id* as such: `'{{resolve:secretsmanager:prod-DatabaseConnParams:SecretString:RDS_HOSTNAME:<version-stage>:<version-id>}}'`. When you do not specify a version, CloudFormation defaults to resolving the secret associated with the stage `AWSCURRENT`.]{type="info"}
```

**Comment by sdangol** [comment]
I was getting an error `Cannot find version 8.0.26 for mysql` when trying to deploy the`cfn-workshop-dynamic-references-database` stack. Can you please check

**Review by sdangol — CHANGES_REQUESTED** [review]
Added some feedbacks

File: content/intermediate/templates/dynamic-references/index.md
Comment (cjhar) [reply content/intermediate/templates/dynamic-references/index.md:1]: Fixed

File: content/intermediate/templates/dynamic-references/index.md (hunk 1 above)
Comment (cjhar) [reply content/intermediate/templates/dynamic-references/index.md:64]: Fixed

File: content/intermediate/templates/dynamic-references/index.md
Comment (cjhar) [reply content/intermediate/templates/dynamic-references/index.md:1]: Fixed

File: content/intermediate/templates/dynamic-references/index.md [hunk 3]
```diff
@@ -131,21 +195,43 @@ Let’s get started! Choose to follow steps shown next:
...
     6. Choose to use default values for **Configure stack options**, and choose **Next**.
-    7. On the **Review** page for your stack, scroll down to the bottom, and select the IAM Capabilities check box as shown in the following example:
-       ![Acknowledge IAM Capability](/static/intermediate/templates/dynamic-references/iam-capability.png)
-    8. Choose **Create** stack. Refresh the page until you see your stack in the `CREATE_COMPLETE` status.
+    7. On the **Review <stack_name>** page, scroll down to the bottom and tick **I acknowledge that AWS CloudFormation might create IAM resources** check box, then click on **Submit**.
+    8. Wait for stack status to reach the **CREATE_COMPLETE**. You need to periodically select Refresh to see the latest stack status.
+    ::::
+    :::::
 
    In the template you just used, database connection parameters are retrieved during stack runtime using a dynamic string. You retrieved the value for a given key, such as `RDS_HOSTNAME`, with: `'{{resolve\:secretsmanager\:DatabaseConnParams\:SecretString\:RDS_HOSTNAME}}'`, where `DatabaseConnParams` is the secret ID.
```
[the remark this answers is not in the document]
Comment (cjhar) [reply content/intermediate/templates/dynamic-references/index.md:230]: Fixed

File: content/intermediate/templates/dynamic-references/index.md (hunk 2 above)
Comment (cjhar) [reply content/intermediate/templates/dynamic-references/index.md:232]: Fixed

**Review by sdangol — APPROVED** [review]
LGTM

**Comment by rezabekf** [comment]
> 

Perfect, I have added lab duration.

**Review by rezabekf — APPROVED** [review]
This was the big one. Thanks all for contributing! LGMT

**Pull request merged**